WoWsShipBuilder.Web/Pages/ShipComparison.razor

```diff
@@ -0,0 +1,694 @@
+        }
+
+        if (data.Any())
+        {
+            foreach (SortDefinition<ShipComparisonDataWrapper>? sortDefinition in state.SortDefinitions)
+            {
+                object? sortItem = sortDefinition.SortFunc.Invoke(data.First());
+                if (sortItem is List<decimal>)
+                {
+                    if (sortDefinition.Descending)
+                    {
+                        data = data.OrderByDescending(x => ((List<decimal>) sortDefinition.SortFunc.Invoke(x)).DefaultIfEmpty().Max()).ToList();
+                    }
+                    else
+                    {
+                        data = data.OrderBy(x => ((List<decimal>) sortDefinition.SortFunc.Invoke(x)).DefaultIfEmpty().Min()).ToList();
+                    }
+                }
+                else if (sortItem is List<int>)
+                {
+                    if (sortDefinition.Descending)
+                    {
+                        data = data.OrderByDescending(x => ((List<int>) sortDefinition.SortFunc.Invoke(x)).DefaultIfEmpty().Max()).ToList();
+                    }
+                    else
+                    {
+                        data = data.OrderBy(x => ((List<int>) sortDefinition.SortFunc.Invoke(x)).DefaultIfEmpty().Min()).ToList();
+                    }
+                }
+                else
+                {
+                    if (sortDefinition.Descending)
+                    {
+                        data = data.OrderByDescending(sortDefinition.SortFunc).ToList();
+                    }
+                    else
+                    {
+                        data = data.OrderBy(sortDefinition.SortFunc).ToList();
+                    }
+                }
+            }
+        }
+
+        return Task.FromResult<GridData<ShipComparisonDataWrapper>>(new() { TotalItems = data.Count, Items = data.Skip(state.Page * state.PageSize).Take(state.PageSize).ToList() });
+    }
+
+    private void UpdateUrl()
+    {
+        var tierFilter = $"tiers={string.Join(',', ViewModel!.SelectedTiers)}";
+        var classFilter = $"classes={string.Join(',', ViewModel!.SelectedClasses)}";
+        var nationFilter = $"nations={string.Join(',', ViewModel!.SelectedNations)}";
+        var categoryFilter = $"categories={string.Join(',', ViewModel!.SelectedCategories)}";
+        
+        string toggleUpgradedModules = ViewModel!.UseUpgradedModules ? "upgraded-modules=true" : string.Empty;
+        string toggleHideUnfittingShips = ViewModel!.HideShipsWithoutSelectedSection ? "hide-unfitting-ships=true" : string.Empty;
+
+        string dataSection = $"section={ViewModel!.SelectedDataSection}";
+
+        NavManager.NavigateTo($"/comparison?{tierFilter}&{classFilter}&{nationFilter}&{categoryFilter}&{toggleUpgradedModules}&{toggleHideUnfittingShips}&{dataSection}");
```

Comment:
Consider putting the page link and the parameter names into constants.
Also consider using QueryBuilder to build the query string, makes your own constructs here redundant.